### DISCUSSION TITLE
Any API to lock/unlock page or database?

### DISCUSSION THREAD

**Discussion by RickVincent** [jheddings/notional#56] [Q&A] [answered]
After update page or database, I am trying to lock it. But haven't found relative API from notional. Any comment? Also need unlock method before update page/database.

BTW, any way to check if a block is BLANK one? Can I determine it by blow conditions?
_Block.has_children is None_ and _len(notion.blocks.retrieve(block_id).first()) is None_

The purpose is to remove all BLANK blocks from a specific page.

**Comment by jheddings** [comment, marked as answer]
Hi @RickVincent 👋 !

> After update page or database, I am trying to lock it. But haven't found relative API from notional. Any comment? Also need unlock method before update page/database.

The API does not currently expose the "lock" abilities...  I assume (perhaps naively) that when it is available, locking will follow a similar pattern as the `archive` flag on the Database block.

> any way to check if a block is BLANK one?

You could iterate over all blocks on the page using a loop like the following...  I haven't had a chance to test this, so YMMV 😬 

```python
# similar to the loop in examples/page.py
page = notion.pages.retrieve(page_id)
for block in notion.blocks.children.list(page):
    if block isinstance(blocks.TextBlock) and block.PlainText == "":
        notion.blocks.delete(block)
```

Since different blocks have a different concept of "blank", you may need to inspect the type of block and use some of its properties to make a decision about being empty (or not).

Hope that helps!

**Reply by RickVincent** [reply]
Hi @jheddings,

Appreciate your quick answer! Just had a try, and your code works perfectly to removed all BLANK blocks!
